src/sdk/pynni/nni/compression/tensorflow/builtin_pruners.py

```diff
@@ -98,3 +98,98 @@ def update_epoch(self, epoch, sess):
         sess.run(tf.assign(self.now_epoch, int(epoch)))
         for k in self.if_init_list:
             self.if_init_list[k] = True
+
+class FPGMPruner(Pruner):
+    """A filter pruner via geometric median.
+    "Filter Pruning via Geometric Median for Deep Convolutional Neural Networks Acceleration", 
+    https://arxiv.org/pdf/1811.00250.pdf
+    """
+
+    def __init__(self, model, config_list):
+        """
+        Parameters
+        ----------
+        model : pytorch model
+            the model user wants to compress
+        config_list: list
+            support key for each list item:
+                - pruning_rate: percentage of convolutional filters to be pruned.
+        """
+        super().__init__(model, config_list)
+        self.mask_list = {}
+        self.assign_handler = []
+
+    def calc_mask(self, layer, config):
+        """supports Conv1d, Conv2d, Conv3d
+        filter dimensions for Conv1D:
+        LEN: filter length
+        IN: number of input channel
+        OUT: number of output channel
+
+        filter dimensions for Conv2D:
+        H: filter height
+        W: filter width
+        IN: number of input channel
+        OUT: number of output channel
+
+        Parameters
+        ----------
+        layer : LayerInfo
+            calculate mask for `layer`'s weight
+        config : dict
+            the configuration for generating the mask
+
+        """
+
+        weight = layer.weight
+        op_type = layer.type
+        op_name = layer.name
+        assert 0 <= config.get('pruning_rate') < 1
+        assert op_type in ['Conv1D', 'Conv2D']
```

Comment:
conv3d? docstring says conv3d is supported

Reply:
Thanks, fixed.